src/renderers/shared/fiber/ReactFiberBeginWork.js

```diff
@@ -436,6 +438,16 @@ module.exports = function<T, P, I, TI, C, CX>(
     } else {
       // Proceed under the assumption that this is a functional component
       workInProgress.tag = FunctionalComponent;
+      if (__DEV__) {
+        warning(
+          !!workInProgress.ref,
+          'Stateless function components cannot be given refs ' +
+          '(See ref "' + workInProgress.ref._stringRef + '" in ' +
```

Comment:
Yeah, I opened this early for feedback (thanks, btw!)

I didn't see a test case for what should happen in the function ref case already. assuming there isn't a current message for that case what *should* it be? I don't think it will be too bad to add that case (I'm fiber at least. Will need to look into amending stack)